SofaKernel/modules/SofaHelper/src/sofa/helper/system/PluginManager.cpp

```diff
@@ -417,11 +429,26 @@ bool PluginManager::pluginIsLoaded(const std::string& plugin)
         if (!FileSystem::isFile(plugin))
         {
             // path is invalid
-            msg_error("PluginManager") << "File not found: " << plugin;
+            msg_error("PluginManager") << "Could not check if the plugin is loaded as the path is invalid: " << plugin;
             return false;
         }
 
         pluginPath = plugin;
+
+        // argument is a path but we need to check if it was not already loaded with a different path
+        const auto& pluginName = sofa::helper::system::SetDirectory::GetFileNameWithoutExtension(pluginPath.c_str());
+        for (const auto& k : m_pluginMap)
+        {
+            if (pluginName == k.second.getModuleName() && pluginPath != k.first)
```

Comment:
```suggestion
        for (const auto& [loadedPath, loadedPlugin] : m_pluginMap)
        {
            if (pluginName == loadedPlugin.getModuleName() && pluginPath != loadedPath)
```
Not sure it's better, but definitively longer...

Reply:
I am not really used to this pythonic way for iterating maps but it is good to know it for my part.
And you dont need to know what k is (aka looking at the map def)
so i am good with this suggestion 😙